pennylane/transforms/split_non_commuting.py

```diff
@@ -243,6 +243,8 @@ def circuit(x):
         [[expval(X(0)), probs(wires=[1])], [probs(wires=[0, 1])]]
 
     """
+    if len(tape.measurements) == 0:
+        return [tape], lambda x: x[0]
```

Comment:
Can you replace the lambda function with a normal function? I think we've generally avoided using lambdas for transform processing functions.

Reply:
pylint will complain, and lambda's can be very difficult to debug and inspect.

Reply:
Updated to use `null_postprocessing`